File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = []
    word_list = [words[0]]
    person = 0
    order = 0
    for i in range(1, len(words)):
        if words[i - 1][-1] != words[i][0] or words[i] in word_list:
            person = (i % n) + 1
            order = (i // n) + 1
            break
        word_list.append(words[i])
    
    answer = [person, order]

    return answer

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        HashSet<String> wordSet = new HashSet<>();
        wordSet.add(words[0]);
        
        for(int i = 1; i < words.length; i ++) {
            char beforeLast = words[i-1].charAt(words[i-1].length() - 1);
            char currentFront = words[i].charAt(0);
            
            if(beforeLast != currentFront) {
                return calculate(n, i);
            }
            boolean isSuccess = wordSet.add(words[i]);
            
            if(!isSuccess) {
                return calculate(n, i);
            }
        }
        
        return new int[]{0,0};
    }
    
    private int[] calculate(int n, int idx) {
        int number = (idx % n) + 1;
        int order = (idx / n) + 1;
        
        return new int[]{number, order};
    }
}